qa/util/src/main/java/io/camunda/qa/util/cluster/NewCamundaTestApplication.java

```diff
@@ -0,0 +1,386 @@
+      throw new IllegalStateException(
+          "Expected to get the gateway address for this broker, but the embedded gateway is not enabled");
+    }
+
+    return TestGateway.super.grpcAddress();
+  }
+
+  @Override
+  public GatewayHealthActuator gatewayHealth() {
+    throw new UnsupportedOperationException("Brokers do not support the gateway health indicators");
+  }
+
+  @Override
+  public NewCamundaTestApplication withGatewayConfig(final Consumer<GatewayCfg> modifier) {
+    modifier.accept(brokerProperties.getGateway());
+    return this;
+  }
+
+  @Override
+  public GatewayCfg gatewayConfig() {
+    return brokerProperties.getGateway();
+  }
+
+  @Override
+  public CamundaClientBuilder newClientBuilder() {
+    if (!isGateway()) {
+      throw new IllegalStateException(
+          "Cannot create a new client for this broker, as it does not have an embedded gateway");
+    }
+
+    return TestGateway.super.newClientBuilder();
+  }
+
+  /** Returns the broker configuration */
+  public BrokerBasedProperties brokerConfig() {
+    return brokerProperties;
+  }
+
+  /**
+   * Modifies the broker configuration. Will still mutate the configuration if the broker is
+   * started, but likely has no effect until it's restarted.
+   */
+  public NewCamundaTestApplication withBrokerConfig(
+      final Consumer<BrokerBasedProperties> modifier) {
+    modifier.accept(brokerProperties);
+    return this;
+  }
+
+  /**
+   * Modifies the security configuration. Will still mutate the configuration if the broker is
+   * started, but likely has no effect until it's restarted.
+   */
+  public NewCamundaTestApplication withSecurityConfig(
+      final Consumer<CamundaSecurityProperties> modifier) {
+    modifier.accept(securityConfig);
+    return this;
+  }
+
+  /**
+   * Enables/disables usage of the recording exporter using {@link #RECORDING_EXPORTER_ID} as its
+   * unique ID.
+   *
+   * @param useRecordingExporter if true, will enable the exporter; if false, will remove it from
+   *     the config
+   * @return itself for chaining
+   */
+  public NewCamundaTestApplication withRecordingExporter(final boolean useRecordingExporter) {
+    if (!useRecordingExporter) {
+      registeredExporters.remove(RECORDING_EXPORTER_ID);
+      return this;
+    }
+    return withExporter(
+        RECORDING_EXPORTER_ID, cfg -> cfg.setClassName(RecordingExporter.class.getName()));
+  }
+
+  /**
+   * Registers or replaces a new exporter with the given ID. If it was already registered, the
+   * existing configuration is passed to the modifier.
+   *
+   * @param id the ID of the exporter
+   * @param modifier a configuration function
+   * @return itself for chaining
+   */
+  public NewCamundaTestApplication withExporter(
+      final String id, final Consumer<ExporterCfg> modifier) {
+    registeredExporters.merge(id, modifier, (key, cfg) -> cfg.andThen(modifier));
+    return this;
+  }
+
+  public NewCamundaTestApplication withElasticsearchSupport(
```

Comment:
To be discussed here https://github.com/camunda/camunda/pull/27109/files#r1920413439